### PR TITLE
[IMP] point_of_sale, pos_hr: open session when load pos

### PR DESCRIPTION
**Pull request by davidmonnom**
Before when trying to access to the POS url with no opened session, the user was redirected to the backend POS dashboard.

Now, when the user access to the POS url with no opened session, the session is opened automatically.

taskId: 4161131

### CODE REVIEW

File: addons/web/static/src/core/confirmation_dialog/confirmation_dialog.js [hunk 1]
```diff
@@ -95,6 +95,7 @@ export class AlertDialog extends ConfirmationDialog {
     static props = {
         ...ConfirmationDialog.props,
         contentClass: { type: String, optional: true },
+        getPayload: { type: Function, optional: true },
```
Comment (caburj) [inline addons/web/static/src/core/confirmation_dialog/confirmation_dialog.js:98]: move this somewhere else. I think this is only used in pos.

Comment (davidmonnom) [reply addons/web/static/src/core/confirmation_dialog/confirmation_dialog.js:98]: We use directly that one in POS, if I move it, I need to create an entire override just to add an optional props

File: addons/point_of_sale/models/pos_session.py [hunk 2]
```diff
@@ -184,6 +184,15 @@ def load_data(self, models_to_load, only_data=False):
 
         return response
 
+    def delete_opening_control_session(self):
+        self.ensure_one()
+        if self.state != 'opening_control':
+            raise UserError(_("You can only cancel a session that is in opening control state."))
+        self.sudo().unlink()
```
Comment (caburj) [inline addons/point_of_sale/models/pos_session.py:191]: check if there are no orders in the session. otherwise, they'll be orphaned.

**Comment by caburj** [comment]
robodoo r+

**Comment by caburj** [comment]
robodoo rebase-ff

**Pull request closed without merging**